### PR TITLE
Remove dependency on `Compat.jl`

### PR DESCRIPTION
**Pull request by DilumAluthge**

### CODE REVIEW

**Comment by nalimilan** [comment]
There are lots of (probably unnecessary) uses under test/.

File: Project.toml [hunk 1]
```diff
@@ -1,18 +1,17 @@
 name = "CategoricalArrays"
 uuid = "324d7699-5711-5eae-9e2f-1d82baa6b597"
-version = "0.7.4"
+version = "0.7.5"
```
Comment (nalimilan) [inline Project.toml]: Please keep the version as-is for now, we'll likely go to 0.8 for the next release.

Comment (DilumAluthge) [reply Project.toml]: Done. I changed the version back to 0.7.4.

File: test/06_show.jl [hunk 2]
```diff
@@ -80,37 +79,72 @@ end
     ov1 = CategoricalValue(1, opool)
     ov2 = CategoricalValue(2, opool)
     ov3 = CategoricalValue(3, opool)
+    
+    if Base.VERSION >= v"1.4.0-DEV.591"
```
Comment (nalimilan) [inline test/06_show.jl]: Don't worry about Julia master, that change is going to be reverted soon (it's been that way for 1.2 and 1.3). Anyway these changes shouldn't be needed for this PR.

**Comment by DilumAluthge** [comment]
For the life of me, I cannot get this PR to work. I guess Compat will stay for now.

**Comment by nalimilan** [comment]
It seems you were very close. I've opened https://github.com/JuliaData/CategoricalArrays.jl/pull/237 to finish this.

**Pull request closed without merging**